paddle/fluid/framework/data_layout_transform.cc

```diff
@@ -106,7 +106,8 @@ using mkldnn::memory;
 using mkldnn::primitive;
 using mkldnn::reorder;
 
-void* GetDataFromTensor(const Tensor& tensor, mkldnn::memory::data_type type) {
+template <class T>
+void* GetDataFromTensor(const Tensor& tensor, T type) {
```

Comment:
Templatization of the function is redundant here.

Reply:
You are right. I removed it.